lib/_http_outgoing.js

```diff
@@ -90,11 +90,12 @@ exports.OutgoingMessage = OutgoingMessage;
 
 OutgoingMessage.prototype.setTimeout = function(msecs, callback) {
 
-  if (callback) {
-    if (typeof callback !== 'function')
-      throw new TypeError('callback must be a function');
+  if (typeof callback === 'function') {
     this.on('timeout', callback);
   }
+  else {
+    throw new TypeError('callback must be a function');
```

Comment:
This is not necessary. The `on` function takes care of it already. I would say you can simply remove the `if` guard around `on` and `once`, as they check if the parameter passed to them are functions.